Update event handler attributes in global attributes

sczesny opened the pull request:
<!-- 🙌 Thanks for contributing to MDN Web Docs. 🙌 -->

<!--
Add details below to help us review your pull request (PR).
Explain your changes and link to a related issue or pull request.
Your PR may be delayed or closed if you don't provide enough information.
-->

### Description

Removed: `onautocomplete`, `onautocompleteerror`, `onshow`, `onsort` due to not being documented anywhere (only in w3 drafts)
Removed: `onabort` due to not being a global event listener (https://html.spec.whatwg.org/multipage/dom.html#global-attributes)

Added: `onafterscriptexecute`, `onanimationcancel`, `onanimationend`, `onanimationiteration`, `onanimationstart`, `onauxclick`, `onbeforeinput`, `onbeforematch`, `onbeforescriptexecute`, `onbeforetoggle`, `onbeforexrselect`, `oncommand`, `oncompositionend`, `oncompositionstart`, `oncompositionupdate`, `oncontentvisibilityautostatechange`, `oncontextlost`, `oncontextrestored`, `oncopy`, `oncut`, `onfocusin`, `onfocusout`, `onformdata`, `onfullscreenchange`, `onfullscreenerror`, `ongesturechange`, `ongestureend`, `ongesturestart`, `ongotpointercapture`, `onlostpointercapture`, `onpaste`, `onpointercancel`, `onpointerdown`, `onpointerenter`, `onpointerleave`, `onpointermove`, `onpointerout`, `onpointerover`, `onpointerrawupdate`, `onpointerup`, `onscrollend`, `onscrollsnapchange`, `onscrollsnapchanging`, `onsecuritypolicyviolation`, `onselectstart`, `onslotchange`, `ontouchcancel`, `ontouchend`, `ontouchmove`, `ontouchstart`, `ontransitioncancel`, `ontransitionend`, `ontransitionrun`, `ontransitionstart`, `onwebkitmouseforcechanged`, `onwebkitmouseforcedown`, `onwebkitmouseforceup`, `onwebkitmouseforcewillbegin`, `onwheel`

Kept: `onblur`, `oncancel`, `oncanplay`, `oncanplaythrough`, `onchange`, `onclick`, `onclose`, `oncontextmenu`, `oncuechange`, `ondblclick`, `ondrag`, `ondragend`, `ondragenter`, `ondragleave`, `ondragover`, `ondragstart`, `ondrop`, `ondurationchange`, `onemptied`, `onended`, `onerror`, `onfocus`, `oninput`, `oninvalid`, `onkeydown`, `onkeypress`, `onkeyup`, `onload`, `onloadeddata`, `onloadedmetadata`, `onloadstart`, `onmousedown`, `onmouseenter`, `onmouseleave`, `onmousemove`, `onmouseout`, `onmouseover`, `onmouseup`, `onmousewheel`, `onpause`, `onplay`, `onplaying`, `onprogress`, `onratechange`, `onreset`, `onresize`, `onscroll`, `onseeked`, `onseeking`, `onselect`, `onstalled`, `onsubmit`, `onsuspend`, `ontimeupdate`, `ontoggle`, `onvolumechange`, `onwaiting`

Moved everything to a list with a new section called **List of global event handler attributes**, which is sorted alphabetically.

### Motivation

The current list is out of date. It contains non-existent and deprecated events.

### Additional details

The new list includes all attributes from html.spec.whatwg.org in addition to deprecated `onmousewheel`, `afterscriptexecute` and `beforescriptexecute` event handlers.

### Related issues and pull requests

<!-- 🔨 If this fully resolves a GitHub issue, use "Fixes #123" -->
<!-- 👉 Highlight related pull requests using "Relates to #123" -->
<!-- ❗ If another pull request must be merged first, use "**Depends on:** #123" -->

<!-- 🔎 After submitting, the 'Checks' tab of your PR shows the build status. -->

sczesny committed:
Update event handler attributes in global attributes

sczesny committed:
Revise event handler attributes in documentation

Removed listeners for `window`.

sczesny commented:
@Josh-Cena You're right. Removed these listeners.

Josh-Cena commented on files/en-us/web/html/reference/global_attributes/index.md:
```diff
@@ -14,7 +14,7 @@ In addition to the basic HTML global attributes, the following global attributes
 
 - `xml:lang` and `xml:base` — these are inherited from the XHTML specifications and deprecated, but kept for compatibility purposes.
 - The ARIA [`role`](/en-US/docs/Web/Accessibility/ARIA/Reference/Roles) attribute and the multiple [`aria-*`](/en-US/docs/Web/Accessibility/ARIA/Reference/Attributes) states and properties, used for ensuring accessibility.
-- The [event handler](/en-US/docs/Web/HTML/Reference/Attributes#event_handler_attributes) attributes: `onabort`, `onautocomplete`, `onautocompleteerror`, `onblur`, `oncancel`, `oncanplay`, `oncanplaythrough`, `onchange`, `onclick`, `onclose`, `oncontextmenu`, `oncuechange`, `ondblclick`, `ondrag`, `ondragend`, `ondragenter`, `ondragleave`, `ondragover`, `ondragstart`, `ondrop`, `ondurationchange` ...
```
You can check the spec link above. `onvolumechange` is a valid attribute (conformance-wise) on all elements, although it only makes sense on media elements.

`onDOMActivate`, `onDOMMouseScroll`, `onMozMousePixelScroll` aren't real attributes AFAIK, but correct me if I'm wrong. Definitely fine to leave out since they aren't in the HTML spec. The `afterscriptexecute` and `beforescriptexecute` events are being removed altogether. `onkeypress` and `onmousewheel` need to stay because they are valid HTML attributes, but we can turn this into a list so you can add `{{deprecated_inline}}` if you really wish (I personally don't think it matters).

sczesny committed:
Add global event handler attributes section

Added a new section listing global event handler attributes with detailed descriptions.

sczesny committed:
Refactor global event handler attributes list

Removed duplicate entries and cleaned up global event handler attributes.

sczesny commented:
@Josh-Cena I moved everything to a **List of global event handler attributes** below and included attributes from [html.spec.whatwg.org](https://html.spec.whatwg.org/multipage/dom.html#global-attributes), `onmousewheel`, `afterscriptexecute` and `beforescriptexecute`.
Let me know if I'm missing something.

sczesny committed:
Update global event handler attributes list

Added: `onabort`, `onselectionchange`
Removed: `onafterscriptexecute`, `onbeforescriptexecute`, `onbeforexrselect`, `oncompositionend`, `oncompositionstart`, `oncompositionupdate`

sczesny committed:
Fix formatting

Josh-Cena commented on files/en-us/web/html/reference/global_attributes/index.md:
```diff
@@ -14,7 +14,7 @@ In addition to the basic HTML global attributes, the following global attributes
 - `xml:lang` and `xml:base` — these are inherited from the XHTML specifications and deprecated, but kept for compatibility purposes.
 - The ARIA [`role`](/en-US/docs/Web/Accessibility/ARIA/Reference/Roles) attribute and the multiple [`aria-*`](/en-US/docs/Web/Accessibility/ARIA/Reference/Attributes) states and properties, used for ensuring accessibility.
-- The [event handler](/en-US/docs/Web/HTML/Reference/Attributes#event_handler_attributes) attributes: `onabort`, `onautocomplete`, `onautocompleteerror`, `onblur`, `oncancel`, `oncanplay`, `oncanplaythrough`, `onchange`, `onclick`, `onclose`, `oncontextmenu`, `oncuechange`, `ondblclick`, `ondrag`, `ondragend`, `ondragenter`, `ondragleave`, `ondragover`, `ondragstart`, `ondrop`, `ondurationchange` ...
+- The [event handler attributes](/en-US/docs/Web/HTML/Reference/Global_attributes#list_of_global_event_handler_attributes) listed below.
```
```suggestion
- The [event handler attributes](#list_of_global_event_handler_attributes) listed below.
```

sczesny committed:
Update files/en-us/web/html/reference/global_attributes/index.md

Co-authored-by: Joshua Chen <sidachen2003@gmail.com>

sczesny committed:
Refactor global event handler attributes list format

Swap to inline representation

sczesny commented on files/en-us/web/html/reference/global_attributes/index.md:
```diff
@@ -120,6 +120,293 @@ In addition to the basic HTML global attributes, the following global attributes
+- [`onwebkitmouseforcewillbegin`](/en-US/docs/Web/API/Element/webkitmouseforcewillbegin_event) {{non-standard_inline}}
+  - : Fired before the [`mousedown`](/en-US/docs/Web/API/Element/mousedown_event) event.
+- [`onwheel`](/en-US/docs/Web/API/Element/wheel_event)
+  - : Fired when the user rotates a wheel button on a pointing device (typically a mouse).
```
Updated the format 👍

Josh-Cena committed:
Reduce text again

Josh-Cena reviewed (approved):
This looks really good now. Thanks for sending this! I just removed all the extra explanation (I probably didn't make this instruction clear) so it's just a simple UL, and added some more intro text.
